contracts/examples/DXswapRelayer.sol

```diff
@@ -0,0 +1,325 @@
+            (uint256 amountA, uint256 amountB, uint256 liquidity) = IDXswapRouter(dxSwapRouter).addLiquidity(
+                _tokenA,
+                _tokenB,
+                _amountA,
+                _amountB,
+                minA,
+                minB,
+                dxdaoAvatar,
+                block.timestamp
+            );
+            emit Pool(_tokenA, _tokenB, amountA, amountB, minA, minB, liquidity);
+        } else {
+            address token = _tokenA == address(0) ? _tokenB : _tokenA;
+            uint256 amount = _tokenA == address(0) ? _amountB : _amountA;
+            uint256 value = _tokenA == address(0) ? _amountA : _amountB;
+            uint256 minToken = _tokenA == address(0) ? minB : minA;
+            uint256 minETH = _tokenA == address(0) ? minA : minB;
+
+            TransferHelper.safeApprove(token, dxSwapRouter, amount);
+            (uint256 amountToken, uint256 amountETH, uint256 liquidity) = IDXswapRouter(dxSwapRouter).addLiquidityETH{
+                value: value
+            }(token, amount, minToken, minETH, dxdaoAvatar, block.timestamp);
+            emit Pool(address(0), token, amountETH, amountToken, minETH, minToken, liquidity);
+        }
+    }
+
+    function getPriceOracleFactory(
+        address tokenA,
+        address tokenB,
+        uint256 minReserveA,
+        uint256 minReserveB
+    ) internal view returns (address factory) {
+        (uint256 dxSwapReserveA, uint256 dxSwapReserveB) = DXswapLibrary.getReserves(dxSwapFactory, tokenA, tokenB);
+        (uint256 uniswapReserveA, uint256 uniswapReserveB) = DXswapLibrary.getReserves(uniswapFactory, tokenA, tokenB);
+        require(
+            (dxSwapReserveA >= minReserveA && dxSwapReserveB >= minReserveB) ||
+                (uniswapReserveA >= minReserveA && uniswapReserveB >= minReserveB),
+            'DXliquidityRelay: INSUFFICIENT_RESERVE'
+        );
+        if (dxSwapReserveA >= minReserveA && dxSwapReserveB >= minReserveB) {
+            factory = dxSwapFactory;
+        } else {
+            factory = uniswapFactory;
+        }
+    }
+
+    function getPoolStake(
+        address factory,
+        address tokenA,
+        address tokenB,
+        uint256 amountA,
+        uint256 amountB
+    ) internal view returns (uint256 poolStake) {
+        (uint256 reserveA, uint256 reserveB) = DXswapLibrary.getReserves(factory, tokenA, tokenB);
+        poolStake = (amountA.add(amountB)) / ((reserveA.add(reserveB))).mul(10000);
+    }
+
+    function consultOracleParameters(
+        IDXswapFactory factory,
+        address tokenA,
+        address tokenB,
+        uint256 amountA,
+        uint256 amountB,
+        uint256 maxWindows
+    ) internal view returns (uint256 windowSize, uint8 granularity) {
+        uint256 poolStake = getPoolStake(address(factory), tokenA, tokenB, amountA, amountB);
+        // poolStake: 0.1% = 10; 1=100; 10% = 1000;
+        uint256 windows;
+        if(poolStake < 10) {
+          windows = 3;
```

Comment:
I think a using switch instead of multiple if conditions will save gas in this case.